modules/n1ql/pages/n1ql-language-reference/createsequence.adoc

```diff
@@ -0,0 +1,414 @@
+
+[source,ebnf]
+----
+include::{partialsdir}/grammar/ddl.ebnf[tag=increment-by]
+----
+
+image::n1ql-language-reference/increment-by.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the INCREMENT BY clause to specify the increment value of each step in the sequence.
+
+If this clause is omitted, the increment value is `1`: that is, at each step in the sequence, the value goes up by `1`.
+
+[horizontal#increment-by-args]
+integer::
+(Required) The step size for the sequence.
+Use a negative value for a descending sequence.
+
+[[maxvalue]]
+==== MAXVALUE Clause
+
+[source,ebnf]
+----
+include::{partialsdir}/grammar/ddl.ebnf[tag=maxvalue]
+----
+
+image::n1ql-language-reference/maxvalue.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the MAXVALUE clause to specify the maximum value for the sequence.
+
+Use NO MAXVALUE to specify that the maximum value is the highest signed 64-bit integer, `2^63^-1`.
+
+If this clause is omitted, the default is NO MAXVALUE.
+
+[horizontal#maxvalue-args]
+integer::
+(Optional) The maximum value for the sequence.
+
+[[minvalue]]
+==== MINVALUE Clause
+
+[source,ebnf]
+----
+include::{partialsdir}/grammar/ddl.ebnf[tag=minvalue]
+----
+
+image::n1ql-language-reference/minvalue.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the MINVALUE clause to specify the minimum value for the sequence.
+
+Use NO MINVALUE to specify that the minimum value is the lowest signed 64-bit integer, `-2^63^`.
+
+If this clause is omitted, the default is NO MINVALUE.
+
+[horizontal#minvalue-args]
+integer::
+(Optional) The minimum value for the sequence.
+
+[[cycle]]
+==== CYCLE Clause
+
+[source,ebnf]
+----
+include::{partialsdir}/grammar/ddl.ebnf[tag=cycle]
+----
+
+image::n1ql-language-reference/cycle.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the CYCLE clause to specify the whether the sequence should begin again when it reaches the maximum or minimum value.
+
+Use NO CYCLE to specify that the sequence should stop when it reaches the maximum or minimum value.
+
+If this clause is omitted, the default is NO CYCLE.
+
+[[cache]]
+==== CACHE Clause
+
+[source,ebnf]
+----
+include::{partialsdir}/grammar/ddl.ebnf[tag=cache]
+----
+
+image::n1ql-language-reference/cache.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the CACHE clause to specify the cache size for the sequence.
+
+Use NO CACHE to specify a cache size of `1`.
+
+If this clause is omitted, the default is NO CACHE.
+
+[horizontal#cache-args]
+integer::
+(Optional) The cache size for the sequence.
+The value must be greater than `0`.
+
+[[sequence-with]]
+=== WITH Clause
+
+[source,ebnf]
+----
+include::{partialsdir}/grammar/ddl.ebnf[tag=sequence-with]
+----
+
+image::n1ql-language-reference/sequence-with.png["Syntax diagram: refer to source code listing", align=left]
+
+Alternatively, use the WITH clause to specify options for the sequence using a JSON object.
```

Comment:
nit: using Alternatively at the start of a section is a little odd. Maybe add more context, 
As an alternative to xref:...#...,

Reply:
Done